### PR TITLE
Add support for Bresser water leakage sensor, PN: 7009975

### PR DESCRIPTION
**Pull request by matthias-bs**

### CODE REVIEW

File: src/devices/bresser_leakage.c [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    // TODO: Find parity/checksum/checksum algorithm
+
+    sensor_id     = ((uint32_t)msg[2] << 24) | (msg[3] << 16) | (msg[4] << 8) | (msg[5]);
+    s_type        = msg[6] >> 4;
+    chan          = (msg[6] & 0x7);
+    battery_ok    = ((msg[7] & 0x30) != 0x00) ? 1 : 0;
+    startup       = (msg[6] >> 3) & 1;
+    alarm         = ((msg[7] & 0x80) == 0x80) ? 1 : 0;
+    no_alarm      = ((msg[7] & 0x40) == 0x40) ? 1 : 0;
+
```
Comment (zuckschwerdt) [inline src/devices/bresser_leakage.c:117]: Preferably don't use double newlines

File: src/devices/bresser_leakage.c [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    for (int i=8; i<=15; i++) {
+        null_bytes |= msg[i];
+    }
+
+    // The parity/checksum/digest algorithm is currently unknown.
+    // We apply some heuristics to validate that the message is really from
+    // a water leakage sensor.
+    decode_ok = (s_type == SENSOR_TYPE_LEAKAGE) &&
+                (alarm != no_alarm) &&
```
Comment (zuckschwerdt) [inline src/devices/bresser_leakage.c:129]: Continued line indent is 8 spaces, not aligned.

File: src/devices/bresser_leakage.c [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            "startup",          "Startup",      DATA_COND,   startup,  DATA_INT,    startup,
+            NULL);
+    /* clang-format on */
+
+    decoder_output_data(decoder, data);
+    return 1;
+}
+
+static char const *const output_fields[] = {
+    "model",
```
Comment (zuckschwerdt) [inline src/devices/bresser_leakage.c:152]: Continued line indent is 8 spaces, not just 4. Also r_device below

File: src/devices/bresser_leakage.c (hunk 1 above)
Comment (matthias-bs) [reply src/devices/bresser_leakage.c:117]: Fixed

File: src/devices/bresser_leakage.c (hunk 2 above)
Comment (matthias-bs) [reply src/devices/bresser_leakage.c]: Fixed

File: src/devices/bresser_leakage.c (hunk 3 above)
Comment (matthias-bs) [reply src/devices/bresser_leakage.c]: Fixed

**Comment by matthias-bs** [comment]
What does "Needs maintainer_update" mean? Is there anything to be done by me?

**Comment by zuckschwerdt** [comment]
> What does "Needs maintainer_update" mean? Is there anything to be done by me?

When the PR is final you should run `./maintainer_update.py` to updates docs and add that commit.

**Comment by matthias-bs** [comment]
Thanks to @MacH-21 for discovering the crc algorithm!

File: README.md [hunk 4]
```diff
@@ -341,7 +342,7 @@ See [CONTRIBUTING.md](./docs/CONTRIBUTING.md).
   [-d <RTL-SDR USB device index>] (default: 0)
   [-d :<RTL-SDR USB device serial (can be set with rtl_eeprom -s)>]
 	To set gain for RTL-SDR use -g <gain> to set an overall gain in dB.
-	SoapySDR device driver is available.
+	SoapySDR device driver is not available.
```
Comment (gdt) [inline README.md:345]: unintended change, also in man page.

**Comment by matthias-bs** [comment]
This happened somehow after running `./maintainer_update.py`.

File: src/devices/bresser_leakage.c [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    alarm         = ((msg[7] & 0x80) == 0x80) ? 1 : 0;
+    no_alarm      = ((msg[7] & 0x40) == 0x40) ? 1 : 0;
+
+    null_bytes = msg[7] & 0xF;
+
+    for (int i=8; i<=15; i++) {
+        null_bytes |= msg[i];
+    }
+
+    // The parity/checksum/digest algorithm is currently unknown.
```
Comment (merbanan) [inline src/devices/bresser_leakage.c:131]: Not true anymore.

Comment (matthias-bs) [reply src/devices/bresser_leakage.c]: Fixed.

**Comment by matthias-bs** [comment]
Is there anything else to be done?

**Comment by matthias-bs** [comment]
Thank you!

**Pull request merged**